src/Symfony/Component/Console/Input/InputOption.php

```diff
@@ -56,8 +56,7 @@ public function __construct($name, $shortcut = null, $mode = null, $description
             if (is_array($shortcut)) {
                 $shortcut = implode('|', $shortcut);
             }
-            $shortcuts = preg_split('{(\|)-?}', ltrim($shortcut, '-'));
-            $shortcuts = array_filter($shortcuts);
+            $shortcuts = array_filter(preg_split('{(\|)-?}', ltrim($shortcut, '-')));
```

Comment:
Not sure about this.

Reply:
No performance gain, just fewer lines of code to maintain.

Reply:
In general, performing two distinct operations in two lines of code is easier to understand than performing two distinct operations in one line of code.

The number of lines of code to be maintained is less of an issue than the ability for the code to be easily understood.

I'd say this specific change is better being reverted to the two line version.